scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,323 @@
+                                                       blocksize*max(m, n))
+        liwork = 8*kmax
+    else:
+        lwork = m + n + 9*kmax + 2*kmax*kmax + 4 + max(m + n, 4*kmax + 4)
+        liwork = 2*kmax + 1
+    work = np.empty(lwork, dtype=typ.lower())
+    iwork = np.empty(liwork, dtype=np.int32)
+
+    # dummy arguments: these are passed to aprod, and not used in this wrapper
+    dparm = np.empty(1, dtype=typ.lower())
+    iparm = np.empty(1, dtype=np.int32)
+
+    if typ.isupper():
+        # PROPACK documentation is unclear on the required length of zwork.
+        # Use the same length Julia's wrapper uses
+        # see https://github.com/JuliaSmoothOptimizers/PROPACK.jl/
+        zwork = np.empty(m + n + 32*m, dtype=typ)
+
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, zwork,
+                                                iwork, doption, ioption,
+                                                dparm, iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, zwork, iwork, doption,
+                                            ioption, dparm, iparm)
+    else:
+        if irl_mode:
+            u, sigma, bnd, v, info = lansvd_irl(_which_converter[which],
+                                                jobu, jobv, m, n,
+                                                shifts, k, maxiter, aprod,
+                                                u, v, tol, work, iwork,
+                                                doption, ioption, dparm,
+                                                iparm)
+        else:
+            u, sigma, bnd, v, info = lansvd(jobu, jobv, m, n, k, aprod, u, v,
+                                            tol, work, iwork, doption,
+                                            ioption, dparm, iparm)
+
+    if info > 0:
+        raise LinAlgError(
+            f"An invariant subspace of dimension {info} was found.")
+    if info < 0:
+        raise LinAlgError(
+            f"k={k} singular triplets did not converge within "
+            f"kmax={kmax} iterations")
+
+    # catch corner case where A is all zeros and v is not orthogonal
+    if not np.any(v):
+        v = np.eye(*v.shape)
+
+    # construct return tuple based on caller's desired output
+    return {
+        (0, 0, 0): sigma,
+        (1, 0, 0): (u[:, :k], sigma),
+        (0, 1, 0): (sigma, v[:, :k].conj().T),
+        (1, 1, 0): (u[:, :k], sigma, v[:, :k].conj().T),
+        (0, 0, 1): (sigma, info, bnd),
+        (1, 1, 1): (u[:, :k], sigma, v[:, :k].conj().T, info, bnd),
+        (0, 1, 1): (sigma, v[:, :k].conj().T, info, bnd),
+        (1, 0, 1): (u[:, :k], sigma, info, bnd),
```

Comment:
`LinAlgError`s are being raises when `info` is non-zero, so can we omit the option to return the `info` option?

Reply:
Yeah I think the Python interface can be Pythonic like this.

Reply:
Removed `info` from return tuple